### PR TITLE
feat: support AuthenticationFilter in GRPCRoute

### PR DESCRIPTION
**Pull request by cnvergence**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes:
- https://github.com/envoyproxy/gateway/issues/1675

### CODE REVIEW

File: internal/provider/kubernetes/routes_test.go [hunk 1]
```diff
@@ -551,6 +552,225 @@ func TestProcessHTTPRoutes(t *testing.T) {
...
+
+	testCases := []struct {
+		name               string
+		routes             []*gwapiv1a2.GRPCRoute
+		authenFilters      []*egv1a1.AuthenticationFilter
+		extensionAPIGroups []schema.GroupVersionKind
+		expected           bool
+	}{
+		{
+			name: "valid httproute",
```
Comment (arkodg) [inline internal/provider/kubernetes/routes_test.go:594]:
```suggestion
			name: "valid grpcroute",
```

File: internal/provider/kubernetes/routes_test.go [hunk 2]
```diff
@@ -551,6 +552,225 @@ func TestProcessHTTPRoutes(t *testing.T) {
...
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "httproute with one authenticationfilter",
```
Comment (arkodg) [inline internal/provider/kubernetes/routes_test.go:637]:
```suggestion
			name: "grpcroute with one authenticationfilter",
```

**Comment by arkodg** [comment]
the PR is looking good, thanks for working on this !
would be good to add one test case in https://github.com/envoyproxy/gateway/tree/main/internal/gatewayapi/testdata

there's a new helper `make testdata` which also generates the testdata out for you :)

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**